guillotina_amqp/commands/worker.py

```diff
@@ -4,22 +4,58 @@
 import aiotask_context
 import asyncio
 import logging
-import sys
-import time
+import threading
+import os
 
 
 logger = logging.getLogger('guillotina_amqp')
 
 
-async def activity_check(worker, timeout):
-    while True:
-        await asyncio.sleep(60)
-        if (time.time() - worker.last_activity) > timeout:
-            logger.error(f'Exiting worker because no activity in {timeout} seconds')
-            sys.exit(1)
+class EventLoopWatchdog(threading.Thread):
```

Comment:
This is clever but part of the reason for `activity_check` was connection issues to rabbitmq.

If I understand this correctly, it will not kill itself if rabbitmq stopped sending the process messages.

Reply:
@vangheem it will kill itself in case the main event loop gets stuck (maybe one of the tasks running is blocking), since the watchdog its running in an independent loop/thread it will continue sensing the loop health, anyway the rmq heartbeat will fail before and the message will be requeued, the goal of the watchdog is kill the stuck worker and force the orchestration manager to create another worker.

Reply:
@dmanchon yes, I understand. The problem is the removal of the previous check that was testing for rabbitmq activity and killing itself if there was none.

Reply:
ah, i see, i didnt understand the goal of that check, since if there were nothing in queue for some time killed the worker, isnt?, immediatly after that our kubernetes spawn a new worker to replace it, so we didnt achieve scaling with that method, plus in case of a blocking task in the event loop that check were never reached and the worker remained indefinetely stuck.

Reply:
"so we didnt achieve scaling with that method" -- I don't understand this. It shouldn't affect scaling at all.

Yes, it would be respawned immediately but it is better than a worker not working and never processing messages.

And yes, I understand the point of the new implementation and it handling a blocked thread. I'm just trying to say removing the old implementation without handling that case in another way wouldn't be good.

Reply:
Sorry but i understood that check as a method to remove a worker when nothing is in the queue, therefore acting as a kind of auto-scaling.

Also, i didnt like the fact that if we are processing a very long running task and there is nothing else in the queue it will kill that job without giving time to finished it. Thats why i added this method that just kill the app in case of a stuck event loop (so blocking all the tasks). 

We can add back that checking as part of the watchdog

Reply:
"Sorry but i understood that check as a method to remove a worker when nothing is in the queue, therefore acting as a kind of auto-scaling." Ahh I see. No, the point was sometimes aioamqp goes wonky and stops receiving.

"Also, i didnt like the fact that if we are processing a very long running task and there is nothing else in the queue it will kill that job without giving time to finished it" Yes, very good point.

"We can add back that checking as part of the watchdog" Awesome!

Reply:
@lferran @vangheem  i propose a beacon message in another queue/exhange (with fanout) to sense if we are receving the messages, if we dont get activity on that queue then we kill the worker, if we receive the beacon then we send it back to a delay queue, that after some time will queue it on the main-beacon queue again.
We can add that on the watchdog logic.
This seems like a corner case, do you think we can address this in a separate PR?

Reply:
Fixing this in another issue sounds good to me, as long as it's ok for you @vangheem

Reply:
Fine with me.

Reply:
:+1: cool, created the issue: https://github.com/guillotinaweb/guillotina_amqp/issues/6